ui/app/components/app/connected-sites-list/connected-sites-list.component.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import Button from '../../ui/button'
+import IconWithFallBack from '../../ui/icon-with-fallback'
+
+export default class ConnectedSitesList extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    renderableDomains: PropTypes.arrayOf(PropTypes.shape({
+      name: PropTypes.string,
+      icon: PropTypes.string,
+      key: PropTypes.string,
+      lastConnectedTime: PropTypes.string,
+      permissionDescriptions: PropTypes.array,
+    })).isRequired,
+    domains: PropTypes.object,
+    showDisconnectAccountModal: PropTypes.func.isRequired,
+    showDisconnectAllModal: PropTypes.func.isRequired,
+  }
+
+  state = {
+    expandedDomain: '',
+    iconError: '',
+    domains: {},
+  }
+
+  handleDomainItemClick (domainKey) {
+    if (this.state.expandedDomain === domainKey) {
+      this.setState({ expandedDomain: '' })
+    } else {
+      this.setState({ expandedDomain: domainKey })
+    }
+  }
+
+  render () {
+    const { renderableDomains, domains, showDisconnectAccountModal, showDisconnectAllModal } = this.props
+    const { expandedDomain } = this.state
+    const { t } = this.context
+
+    return (
+      <div className="connected-sites-list">
+        {
+          renderableDomains.map((domain, domainIndex) => {
+            const domainIsExpanded = expandedDomain === domain.key
+            return (
+              <div
+                className={classnames('connected-sites-list__domain', {
+                  'connected-sites-list__domain--expanded': domainIsExpanded,
+                })}
+                key={`connected-domain-${domainIndex}`}
+              >
+                <div className="connected-sites-list__domain-item" onClick={ () => this.handleDomainItemClick(domain.key) }>
+                  <div className="connected-sites-list__domain-item-info-container">
+                    <IconWithFallBack icon={domain.icon} name={domain.name} />
+
+                    <div className="connected-sites-list__domain-info">
+                      <div className="connected-sites-list__domain-names">
+                        <div className="connected-sites-list__domain-name">
+                          { domain.name }
+                        </div>
+                      </div>
+                      { domain.lastConnectedTime
+                        ? <div className="connected-sites-list__domain-last-connected">
+                          { t('domainLastConnect', [domain.lastConnectedTime]) }
+                        </div>
+                        : null
+                      }
+                      {domainIsExpanded
+                        ? <div className="connected-sites-list__domain-origin">
+                          { domain.key }
```

Comment:
External extension hostnames were shown differently to explain that they were extensions (otherwise it just looks like a nonsense hostname).
The format used previously is `Extension ID: [extensionId]`

Reply:
4a5fecb2f